src/vni-manager.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * File:  vni-manager.js
+ */
+var _ = require('underscore');
+
+var inputStates = require('./input-states');
+
+var IIP_VNID = '';
+
+/**
+ * Manage the node's vni's by vnid.  If no vnid is provided the IIP VNID will be used.
+ * If the vni for the specified vnid does not exist, a new one will be created.
+ *
+ * @this the component node instance unless otherwise specified on an API
+ * @param facade a component facade
+ * @return the component facade, extended to include the vni interface
+ * 
+ * Usage examples: 
+ *     var vni = this.nodeInstance.vni(); // retrieve an IIP vni 
+ *     var vni = this.nodeInstance.vni(vnid); // retrieve the vni associated with the vnid
+ *     var context = component.deleteVnis();
+ *     var context = component.deleteVni( vnid );
+ *     var context = nodeInstance.vni.delete();
+ *     var context = nodeInstance.vni().errorState( state );
+ *     var state = nodeInstance.vni().errorState();
+ *     var context = nodeInstance.vni().inputStates( {input: state} );
+ *     var inputStates = nodeInstance.vni().inputStates();
+ *     var context = nodeInstance.vni().outputState( state );
+ *     var state = nodeInstance.vni().outputState();
+ *     var context = nodeInstance.vni.delete();
```

Comment:
David had a requirement that we be able to delete a VNI.  I have provided two APIs to do it, though both share the same code underneath.  This one will delete the vni from the vnis array.